File: gp-service/src/main/java/fact/it/gpservice/dto/GpResponse.java
package fact.it.gpservice.dto;

import java.time.LocalDate;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GpResponse {
    private Long id;
    private String gpCode;
    private String name;
    private String country;
    private String city;
    private Integer distanceMeters;
    private Integer laps;
    private DriverResponse winningDriver;
    private DriverResponse secondDriver;
    private DriverResponse thirdDriver;
    private TeamResponse winningTeam;
    private LocalDate raceDate;
}

File: formula1_frontend/src/dto/gpResponse.ts
import DriverResponse from "./driverResponse";
import TeamResponse from "./teamResponse";

interface GpResponse {
  gpCode: string;
  name: string;
  country: string;
  city: string;
  distanceMeters: number;
  laps: number;
  winningDriver: DriverResponse;
  secondDriver: DriverResponse;
  thirdDriver: DriverResponse;
  winningTeam: TeamResponse;
  raceDate: Date;
  imageUrl: string;
}

export default GpResponse;
